docs/docs/workflows/s3proxy.md

```diff
@@ -0,0 +1,56 @@
+# Install s3proxy
+
+Constellation includes a transparent encryption proxy for [AWS S3](https://aws.amazon.com/de/s3/).
+s3proxy will encrypt/decrypt objects when you send/retrieve objects to/from S3, without making changes to your application necessary.
+
+## Limitations
+
+:::caution
+Using s3proxy outside Constellation is insecure as the connection between the key management service (KMS) and s3proxy is protected by Constellation's WireGuard VPN.
+The VPN is a feature of Constellation and won't be present by default in other environments.
+:::
+
+These limitations will be removed with future iterations of s3proxy.
+
+- Only `PutObject` and `GetObject` requests are encrypted/decrypted by s3proxy.
+That means other requests are forwarded to S3 as-is.
+For example, objects uploaded with `CreateMultipartUpload` aren't encrypted.
```

Comment:
can we change the behavior so that calls that would expose unencrypted object data fail instead?

Reply:
Recap of phone conversation: decided to implement `insecure` flag for now that allows users to trigger the current behavior. By default these requests will be blocked.

Reply:
#2420 adds the discussed flag. Rephrased section to reflect new situation.